### PR TITLE
S3 PART 3 - Retrieve objects in web!

### PR DESCRIPTION
**Pull request by colegottdank**
- PR returns signed urls in getRequests
- PR updates web hooks to retrieve S3 objects after being returned the signed url

### CODE REVIEW

File: valhalla/jawn/src/lib/shared/request/request.ts [hunk 1]
```diff
@@ -205,7 +214,7 @@ async function mapLLMCalls(
   const promises =
     heliconeRequests?.map(async (heliconeRequest) => {
       // First retrieve s3 signed urls if past the implementation date
-      const s3ImplementationDate = new Date("2024-03-22T23:00:00Z");
+      const s3ImplementationDate = new Date("2024-03-25T21:00:00Z");
```
Comment (chitalian) [inline valhalla/jawn/src/lib/shared/request/request.ts:217]: noice

File: web/lib/api/db/s3Client.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+export class S3Client {
+  private region = "us-west-2";
+  private awsClient: AwsS3Client;
+
+  constructor(
+    accessKey: string,
+    secretKey: string,
+    endpoint: string,
```
Comment (chitalian) [inline web/lib/api/db/s3Client.ts:33]: Why does this need to be passed in if they are the same global variable every time?

File: web/lib/api/graphql/helpers/mappers.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { LlmSchema } from "../../models/requestResponseModel";
+import { HeliconeRequest } from "../../request/request";
+
+export function mapGeminiPro(
```
Comment (chitalian) [inline web/lib/api/graphql/helpers/mappers.ts:4]: this is hillarious. But also probably doesnt belong in this PR. 

It's fine. just keep it in. just saying it 🤷

File: web/lib/api/request/request.ts [hunk 4]
```diff
@@ -201,27 +198,44 @@ export async function getRequestsCached(
...
 async function mapLLMCalls(
   heliconeRequests: HeliconeRequest[] | null,
-  rosettaWrapper: RosettaWrapper
+  s3Client: S3Client,
+  orgId: string
 ): Promise<Result<HeliconeRequest[], string>> {
   const promises =
     heliconeRequests?.map(async (heliconeRequest) => {
+      // First retrieve s3 signed urls if past the implementation date
+      const s3ImplementationDate = new Date("2024-03-25T21:00:00Z");
```
Comment (chitalian) [inline web/lib/api/request/request.ts:221]: This date is used in a few places. Can we hard code this as a constant somewhere? just incase we need to change it in the future?

File: web/services/hooks/requests.tsx [hunk 5]
```diff
@@ -46,7 +46,32 @@ const useGetRequest = (requestId: string) => {
...
+              request: string;
+              response: string;
+            };
+            request.request_body = content.request;
+            request.response_body = content.response;
+          }
+        } catch (error) {
+          console.log(`Error fetching content: ${error}`);
+        }
+      }
```
Comment (chitalian) [inline web/services/hooks/requests.tsx:72]: TODO @ScottMktn we should Lazy load this in the frontend.

Comment (chitalian) [reply web/services/hooks/requests.tsx:72]: in a future PR**

We don't want to wait for all the S3 calls to finish before returning the the frontend. We can return all the other fields for now and then lazy load the bodies in after the fact

File: web/lib/api/graphql/helpers/mappers.ts (hunk 3 above)
Comment (colegottdank) [reply web/lib/api/graphql/helpers/mappers.ts:4]: It is copied from jawn. It just doesn't exist in web atm so wanted to make it the same

File: web/lib/api/request/request.ts (hunk 4 above)
Comment (colegottdank) [reply web/lib/api/request/request.ts:221]: It is used in 2 places: web and jawn. We have no shared repo.

File: web/lib/api/db/s3Client.ts (hunk 2 above)
Comment (colegottdank) [reply web/lib/api/db/s3Client.ts:33]: Either way can work. Passing in does not couple it to the environment variables allowing you to specify on initialization.

File: web/lib/api/request/request.ts (hunk 4 above)
Comment (chitalian) [reply web/lib/api/request/request.ts:221]: rip

**Pull request merged**